_posts/2019-08-26-how-to-use-gruntwork-service-catalog.adoc

```diff
@@ -0,0 +1,1934 @@
+| READMEs, wikis, Slack
+
+| Tests
+| Write automated tests for your infrastructure code. Run tests after every commit and nightly.
+| Terratest
+|===
+
+Most other collections of infrastructure code and service catalogs (e.g., AWS Quick Starts, Bitnami Application Catalog,
+the Terraform Registry, Ansible Galaxy, Chef Supermarket, etc) are useful for learning and example code, but they do
+not take most of this checklist into account, and therefore are not a good fit for direct production use. On the other
+hand, every module in the Gruntwork Service Catalog goes through the production-grade checklist and is explicitly
+designed for use directly in production.
+
+=== Infrastructure as code
+
+Everything in the Gruntwork Service Catalog is designed to allow you to define your _infrastructure as code (IaC)_.
+That is, instead of deploying infrastructure _manually_ (e.g., by clicking around a web page), the idea behind IaC is
+to write code to define, provision, and manage your infrastructure. This has a number of benefits:
+
+Self-service::
+  Most teams that deploy code manually have a small number of sysadmins (often, just one) who are the only ones who
+  know all the magic incantations to make the deployment work and are the only ones with access to production. This
+  becomes a major bottleneck as the company grows. If your infrastructure is defined in code, then the entire
+  deployment process can be automated, and developers can kick off their own deployments whenever necessary.
+
+Speed and safety::
+  If the deployment process is automated, it'll be significantly faster, since a computer can carry out the deployment
+  steps far faster than a person; and safer, since an automated process will be more consistent, more repeatable, and
+  not prone to manual error.
+
+Documentation::
+  Instead of the state of your infrastructure being locked away in a single sysadmin's head, you can represent the
+  state of your infrastructure in source files that anyone can read. In other words, IaC acts as documentation,
+  allowing everyone in the organization to understand how things work, even if the sysadmin goes on vacation.
+
+Version control::
+  You can store your IaC source files in version control, which means the entire history of your infrastructure is now
+  captured in the commit log. This becomes a powerful tool for debugging issues, as any time a problem pops up, your
+  first step will be to check the commit log and find out what changed in your infrastructure, and your second step may
+  be to resolve the problem by simply reverting back to a previous, known-good version of your IaC code.
+
+Validation::
+  If the state of your infrastructure is defined in code, then for every single change, you can perform a code review,
+  run a suite of automated tests, and pass the code through static analysis tools, all practices that are known to
+  significantly reduce the chance of defects.
```

Comment:
Maybe worth quick example of a PR containing some terraform change and the plan output in the PR text?

Reply:
Good idea. Added a TODO, as I'm too lazy to make that happen right now 😁